### PR TITLE
[fastserde] Fix issue with generating schema fingerprint when NaN is default value.

### PR DESCRIPTION
**Pull request by krisso-rtb**

### CODE REVIEW

File: helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/Jackson1Utils.java [hunk 1]
```diff
@@ -210,20 +210,19 @@ public static boolean isRoundNumber(JsonNode node) {
...
     JsonNode defaultValue = field.defaultValue();
-    BigDecimal numericValue = defaultValue.getDecimalValue();
     Schema schema = field.schema();
     // a default value for a union, must match the first element of the union
     Schema.Type defaultType = schema.getType() == UNION ? schema.getTypes().get(0).getType() : schema.getType();
 
     switch (defaultType) {
       case INT:
-        if (!isAMathematicalInteger(numericValue)) {
+        if (!isAMathematicalInteger(defaultValue.getDecimalValue())) {
```
Comment (FelixGV) [inline helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/Jackson1Utils.java:219]: Is this the fix? I don't understand... isn't the code equivalent?

Comment (krisso-rtb) [reply helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/Jackson1Utils.java:219]: If `defaultType` is `DOUBLE` or `FLOAT` and `genericDefaultValue` represents `NaN` then calling `genericDefaultValue.decimalValue()` is
- not needed
- throws `NumberFormatException`

In other words it's called too early.

Comment (FelixGV) [reply helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/Jackson1Utils.java:219]: Oooh I see. Ok, cool. Thanks for explaining, good catch.

**Review by FelixGV — APPROVED** [review]
Thanks!

**Pull request merged**